### DISCUSSION TITLE
コミットする単位について

### DISCUSSION THREAD

**Discussion by ghost** [lgtm-factory/lgtm-factory#24] [未分類]
毎回コミットするタイミングで悩むのですが
皆さんはどれくらいの粒度でコミットしてますか？
[参考](https://qiita.com/jnchito/items/40e0c7d32fde352607be)

（あまりプルリクって送ったことがないので、このdiscus自体無駄だったらすいません🙇）

**Comment by siso001** [comment]
自分は後から修正箇所に気づくことがが多いのでその度にコミットしてますね！

**Reply by ghost** [reply]
@siso001 
自分もこのパターンが多いです！
コミットした後無限に修正しまくってるのでちょっと不安でしたが...

**Reply by kagomen** [reply]
自分も全く同じなんですが、コミット後に内容を修正するコマンドもあるみたいなので、共同開発においては修正コミットを詰み上げるよりかはコミット内容を修正した方が、レビュワーの方が楽になりそうです！
直前のコミット修正くらいだったら、かなり簡単にできるみたいです🌟
https://tech-blog.rakus.co.jp/entry/20191113/git

プッシュしたあとの変更は、結構難しそうです💦

**Reply by siso001** [reply]
コミット修正のコマンドがあるんですね！これからは修正の際にそれを使おうかと思います😃

**Reply by ghost** [reply]
@kagomen 勉強になりました...！！！！！！！

**Comment by kazzyfrog** [comment]
自分は、基本的には、タスク（ステップ）毎に、コミットする感じです！

例えば、自分がこれから着手するイシューの #15 の場合は、
`CODE_OF_CONDUCT.md`ファイルを作成し、
テンプレートに沿って、行動規範を記載するだけなので、１つのコミットかなと思います！


別のケースで、
「お問い合わせフォームの実装」というイシューがあった場合は、
- フォームライブラリ（例えばreact-hook-form）の導入
- 機能実装
- UI、、、

などで、複数のステップ？に分けるかなーと思います！！👍

**Reply by kazzyfrog** [reply]
なので、イシューの中で、以下のように、

- [ ] まずAする
- [ ] 次Bする
- [ ] C、、

ステップというか、細かいタスクを、
あらかじめ明記できると、わかりやすくなるかな感じです！
それに沿って、コミットするようなイメージ！

**Reply by ghost** [reply]
@kazzyfrog 
なるほど！！！細かいタスクにわけてさらにその中でも
ステップでコミットしてく...！このやり方たしかに俯瞰的に物事もみれて
次何しなきゃいけないなーとか自分の進捗状況も把握しやすいのでいいですね！

今回はちょっとこのやり方で粒度小さめでコミットしていこうかなと思います！
